cookbooks/tile/templates/default/export.erb

```diff
@@ -54,6 +56,112 @@ def output_error(message, status = "400 Bad Request"):
   print("</body>")
   print("</html>")
 
+# Add a copyright notice for raster formats (PNG, JPEG, WEBP)
+def add_copyright_notice_raster(image, map_width, map_height, format):
+  # Convert the Mapnik image to PNG and store it in a BytesIO object
+  png = image.tostring("png")
+  png_io = io.BytesIO(png)
+
+  # Load the PNG data from the BytesIO object into a Cairo ImageSurface
+  surface = cairo.ImageSurface.create_from_png(png_io)
+
+  add_copyright_notice_vector(surface, map_width, map_height)
+
+  # Convert the Cairo surface to PNG in a BytesIO object
+  output_io = io.BytesIO()
+  surface.write_to_png(output_io)
+
+  if format == "png":
+    output_io.seek(0)
+    return output_io.read()
```

Comment:
Why are we reading in the data rather than just returning the handle and letting the called use `output_file` on it? If you do that you don't need the seek either as `output_file` already does that.

Reply:
The current raster code doesn’t use output_file either (but sys.stdout.buffer.write(image)). I wanted to keep the current code as is. Using output_code would require some attribution dependent logic in line 135. I’m not sure this is better.

Reply:
I've created a new function `output_image(file)` to avoid some of the code duplication.  `output_file` doesn't seem to work with BytesIO, and` len(image)`  in lines 128+130+132 wouldn't work with an ImageIO object either.

Maybe I'm misunderstanding what exactly you're suggesting. If you have some specific changes in mind, it's probably more efficient if you could propose them as code.

Reply:
I have changed the code accordingly and also added a new function bytesio_size for the payload length.